homeassistant/components/faadelays/binary_sensor.py

```diff
@@ -0,0 +1,104 @@
+"""Platform for FAA Delays sensor component."""
+from homeassistant.components.binary_sensor import BinarySensorEntity
+
+from .const import DOMAIN, FAA_BINARY_SENSORS
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up a FAA sensor based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    binary_sensors = []
+    for kind, attrs in FAA_BINARY_SENSORS.items():
+        name, icon = attrs
+        binary_sensors.append(
+            FAABinarySensor(coordinator, kind, name, icon, entry.entry_id)
+        )
+
+    async_add_entities(binary_sensors, True)
+
+
+class FAABinarySensor(BinarySensorEntity):
```

Comment:
The DataUpdateCoordinator helper also provides a CoordinatorEntity, which handles things like `async_added_to_hass`, `async_update`, `available`, `should_poll`. If this class would inherit from that, all those methods could be removed from here.

Reply:
Fixed half of this, forgot the second half. More incoming.

Reply:
Inherited `async_added_to_hass`, `async_update`, and `available`, but the class wasn't using `should_poll` unless I'm missing it somewhere.